File: Class/final_exam/leap_year.c
#include <stdio.h>
int main()
{
    int year, isLeapYear;
    printf("Enter a year\t");
    scanf("%d", &year);
    if (year % 400 == 0)
        isLeapYear = 1;
    else if (year % 100 == 0)
        isLeapYear = 0;
    else if (year % 4 == 0)
        isLeapYear = 1;
    else
        isLeapYear = 0;
    printf("The year %d is %s\n", year, isLeapYear == 1 ? "Leap Year" : "not a Leap Year");
    return 0;
}

File: Set-2/LeapYear.java

import java.util.Scanner;

class CheckLeapYear {
    public CheckLeapYear() {
        Scanner scan = new Scanner(System.in);
        System.out.print("Enter a year: ");
        int year = scan.nextInt();
        IsLeapYear(year);
    }

    void IsLeapYear(int year) {
        int isLeapYear;
        if (year % 400 == 0)
            isLeapYear = 1;
        else if (year % 100 == 0)
            isLeapYear = 0;
        else if (year % 4 == 0)
            isLeapYear = 1;
        else
            isLeapYear = 0;
        System.out.println(year + " is " + (isLeapYear == 1 ? "a" : "not a") + " leap year");
    }
}

public class LeapYear {
    public static void main(String args[]) {
        new CheckLeapYear();
    }
}
